mira/metamodel/templates.py

```diff
@@ -129,8 +132,11 @@ def with_context(self, do_rename=False, **context) -> "Concept":
         if do_rename:
             if self._base_name is None:
                 self._base_name = self.name
-            name = '_'.join([self._base_name] + \
-                [str(v.replace(':', '_')) for _, v in sorted(context.items())])
+            name_list = [self._base_name]
+            for k, v in sorted(context.items()):
+                nv = name_map.get(v, v) if name_map else v
```

Comment:
maybe we should add a `is_curie` function somewhere that can be used in here to make this code more obvious what it's doing